### PR TITLE
feat: add useBlocksuiteEditor hooks

### PR DESCRIPTION
**Pull request by JimmFly**

### CODE REVIEW

**Comment by JimmFly** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5366** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5366?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5366?utm_source=stack-comment).

**Review by EYHN — APPROVED** [review]

**Review by CatsJuice — APPROVED** [review]

File: packages/frontend/hooks/src/use-block-suite-editor.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { AffineEditorContainer } from '@blocksuite/presets';
+import { atom, useAtom } from 'jotai';
+import { useCallback, useMemo } from 'react';
+
+export const editorContainerAtom = atom<{
```
Comment (pengx17) [inline packages/frontend/hooks/src/use-block-suite-editor.ts]: I think if you expose hooks, there is no point of exposing the atom any longer.
Also the API does not consider the case that two editors share the same pageId

File: packages/frontend/hooks/src/use-block-suite-editor.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        [pageId]: editorContainer,
+      }));
+    },
+    [setEditors, pageId]
+  );
+
+  return useMemo(() => {
+    return {
+      editor,
+      setEditorContainer,
```
Comment (pengx17) [inline packages/frontend/hooks/src/use-block-suite-editor.ts]: the name of the return value may use `useState` pattern that returns a 2-tuple.
also the name of the first param should be `editorContainer`

File: packages/frontend/component/src/components/block-suite-editor/index.tsx [hunk 3]
```diff
@@ -151,7 +148,7 @@ const BlockSuiteEditorImpl = ({
 }: EditorProps): ReactElement => {
   usePageRoot(page);
 
-  const [, setEditorContainer] = useAtom(editorContainerAtom);
+  const { setEditorContainer } = useBlocksuiteEditor(page.id);
```
Comment (pengx17) [inline packages/frontend/component/src/components/block-suite-editor/index.tsx]: Should remove the editor instance when unmount

**Review by pengx17 — CHANGES_REQUESTED** [review]
Please resolve my comments

**Comment by pengx17** [comment]
So the new impl does not support multiple editors?

**Comment by JimmFly** [comment]
> So the new impl does not support multiple editors?

Apologies, it seems I’ve misconstrued your meaning. I interpreted the comment as suggesting that the editor atom didn’t require correspondence with the pageid. @pengx17 

>Also the API does not consider the case that two editors share the same pageId

**Review by pengx17 — APPROVED** [review]

**Pull request merged**